lib/secrets.rb

```diff
@@ -163,6 +195,21 @@ def read_from_vault(key)
     @vault_v2 ? result.data.fetch(:data).fetch(:vault) : result.data.fetch(:vault)
   end
 
+  def write_to_vault(path, data = {})
+    begin
+      result = with_retries { Vault.logical.write(path, data) }
+    rescue Vault::HTTPClientError
+      $!.message.prepend "Error writing to #{key}\n"
+      raise
+    end
+
+    if !result.respond_to?(:data) || !result.data || !result.data.is_a?(Hash)
```

Comment:
nil is not a hash, so can simplify this
```suggestion
    if !result.respond_to?(:data) || !result.data.is_a?(Hash)
```